brian2/core/core_preferences.py

```diff
@@ -14,16 +14,14 @@ def dtype_repr(dtype):
 
 
 def default_float_dtype_validator(dtype):
-    return dtype is float64
+    return dtype in [float32, float64]
```

Comment:
I don't think we need it as a priority, but I wonder how much work it would be to allow for other types such as float16 or float128?

Reply:
I don't think we'd need any additional changes for that, it's just more effort for testing. But we could certainly think about it again when a use case arises.